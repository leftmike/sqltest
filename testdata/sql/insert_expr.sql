--
-- Test INSERT INTO expressions
--

DROP TABLE IF EXISTS tbl1;

CREATE TABLE tbl1 (c1 int, c2 int, c3 int);

INSERT INTO tbl1 VALUES
    (1 + 2, 3 * 4, 6 / 2),
    (12 * 6 + 3 * 2, 12 * 4 + 3, 12 + 4 * 3),
    (12 * (6 + 3), (12 + 4) * 3, 8 / 17);

SELECT * FROM tbl1;

INSERT INTO tbl1 (c1, c3) VALUES (1, 2), (3, 4), (5, 6), (7, 8);

SELECT * FROM tbl1;

DROP TABLE IF EXISTS tbl2;

CREATE TABLE tbl2 (c1 int);

INSERT INTO tbl2 VALUES
    (12 + 34),
    (1234 & 5678),
    (1234 | 5678),
    (4567 / 89),
    (1 << 23),
    (45 % 13),
    (123 * 45),
    (123 - 4567),
    (123456 >> 7),
    (- 123),
    (abs(123)),
    (abs(-123));

SELECT * FROM tbl2;
