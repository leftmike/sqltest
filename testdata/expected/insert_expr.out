--
-- Test INSERT INTO expressions
--
DROP TABLE IF EXISTS tbl1;
CREATE TABLE tbl1 (c1 int, c2 int, c3 int);
INSERT INTO tbl1 VALUES
    (1 + 2, 3 * 4, 6 / 2),
    (12 * 6 + 3 * 2, 12 * 4 + 3, 12 + 4 * 3),
    (12 * (6 + 3), (12 + 4) * 3, 8 / 17);
SELECT * FROM tbl1;
    c1 c2 c3
    -- -- --
 1 108 48  0
 2   3 12  3
 3  78 51 24
(3 rows)
INSERT INTO tbl1 (c1, c3) VALUES (1, 2), (3, 4), (5, 6), (7, 8);
SELECT * FROM tbl1;
    c1 c2 c3
    -- -- --
 1   1     2
 2 108 48  0
 3   3     4
 4   3 12  3
 5   5     6
 6   7     8
 7  78 51 24
(7 rows)
DROP TABLE IF EXISTS tbl2;
CREATE TABLE tbl2 (c1 int);
INSERT INTO tbl2 VALUES
    (12 + 34),
    (1234 & 5678),
    (1234 | 5678),
    (4567 / 89),
    (1 << 23),
    (45 % 13),
    (123 * 45),
    (123 - 4567),
    (123456 >> 7),
    (- 123),
    (abs(123)),
    (abs(-123));
SELECT * FROM tbl2;
         c1
         --
  1    -123
  2   -4444
  3    1026
  4     123
  5     123
  6      46
  7      51
  8    5535
  9    5886
 10       6
 11 8388608
 12     964
(12 rows)
